internal/txprocessor/txprocessor.go

```diff
@@ -80,48 +85,62 @@ func (r TXProcessor) submitTxWithProofs(
 	proofStart := time.Now()
 	hash := hex.EncodeToString(tmtypes.Tx(block.Tx.Data).Hash())
 	neutronTxHash, err := r.submitter.SubmitTxProof(ctx, queryID, block)
-	if err != nil {
-		neutronmetrics.AddFailedProof(string(neutrontypes.InterchainQueryTypeTX), time.Since(proofStart).Seconds())
-		errSetStatus := r.storage.SetTxStatus(
-			queryID, hash, neutronTxHash, relay.SubmittedTxInfo{Status: relay.ErrorOnSubmit, Message: err.Error()})
-		if errSetStatus != nil {
-			return fmt.Errorf("failed to store tx: %w", errSetStatus)
+
+	if err == nil {
+		neutronmetrics.AddSuccessProof(string(neutrontypes.InterchainQueryTypeTX), time.Since(proofStart).Seconds())
+		err = r.storage.SetTxStatus(queryID, hash, neutronTxHash, relay.SubmittedTxInfo{
+			Status: relay.Submitted,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to store tx: %w", err)
 		}
 
-		return fmt.Errorf("could not submit proof for %s with query_id=%d: %w",
-			neutrontypes.InterchainQueryTypeTX, queryID, err)
+		// We submit the PendingSubmittedTxInfo only after checkSubmittedTxStatusDelay to reduce the possibility of
```

Comment:
I think it's now reasonable to move the comment to `delayedTxStatusCheck` description